superset-frontend/packages/superset-ui-core/src/query/types/Query.ts

```diff
@@ -39,12 +39,12 @@ export type QueryObjectFilterClause = {
 } & (
   | {
       op: BinaryOperator;
-      val: string | number | boolean;
+      val: string | number | boolean | null | Date;
```

Comment:
Why use Date and null object on the filter?

Reply:
Because the values come from the `labelMap`:

```
interface EChartTransformedProps<F> {
  labelMap: Record<string, DataRecordValue[]>;
  ...
}

type DataRecordValue = number | string | boolean | Date | null;
```